core/src/main/java/org/apache/accumulo/core/metrics/MetricsUtil.java

```diff
@@ -121,10 +123,29 @@ public static List<Tag> getCommonTags() {
     return commonTags;
   }
 
-  // Centralize any specific string formatting for metric names and/or tags.
+  /**
+   * Centralize any specific string formatting for metric names and/or tags. Ensure strings match
+   * the micrometer naming convention.
+   */
   public static String formatString(String name) {
-    // Ensure strings are not split by the metrics registry naming scheme.
-    // Add special character eliminations here.
+
+    // Handle spaces
+    name = name.replace(" ", ".");
+    // Handle snake_case notation
+    name = name.replace("_", ".");
+
+    // Handle camelCase notation
+    Pattern camelCasePattern = Pattern.compile("[a-z][A-Z][a-z]");
```

Comment:
This compiled pattern could be moved to a static var so its not compiled for each method call.